apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/EphemeralTransaction.cs

```diff
@@ -38,48 +39,41 @@ internal EphemeralTransaction(SpannerConnection connection, TransactionOptions t
 
         public Task<long> ExecuteDmlAsync(ExecuteSqlRequest request, CancellationToken cancellationToken, int timeoutSeconds)
         {
-            return ExecuteHelper.WithErrorTranslationAndProfiling(Impl, "EphemeralTransaction.ExecuteDmlAsync", _connection.Logger);
+            return ExecuteInReadWriteTransactionWithErrorTranslationAndProfiling(Impl, cancellationToken, "EphemeralTransaction.ExecuteDmlAsync");
 
-            async Task<long> Impl()
+            async Task<long> Impl(SpannerTransaction transaction, Action rollbackCheckpoint)
             {
-                using (var transaction = await _connection.BeginTransactionImplAsync(_transactionOptions, TransactionMode.ReadWrite, cancellationToken).ConfigureAwait(false))
+                // TODO: Validate this. Do we really never need to roll back the transaction here?
```

Comment:
Might depend on partioned vs. non-partioned?

Reply:
Maybe. It's frankly unclear to me.

Reply:
Oooh this is something I'm not confident to comment on. I will have to defer to @snehashah16, especially given that this is something that the emulator is expecting but is not an issue on prod.